silkrpc/core/estimate_gas_oracle_test.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+    Copyright 2021 The Silkrpc Authors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "estimate_gas_oracle.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+#include <asio/co_spawn.hpp>
+#include <asio/thread_pool.hpp>
+#include <asio/use_future.hpp>
+#include <boost/endian/conversion.hpp>
+#include <catch2/catch.hpp>
+#include <evmc/evmc.hpp>
+
+#include <silkrpc/types/block.hpp>
+
+namespace silkrpc::ego {
+
+using Catch::Matchers::Message;
+
+TEST_CASE("estimate gas") {
```

Comment:
Clang coverage fails on this PR so you should try to comment out one test at a time to see where the problem is